Makefile

```diff
@@ -301,32 +312,69 @@ ci-build-windows: ensure-release-dir
 ensure-release-dir:
 	mkdir -p $(RELEASE_DIR)
 
+.PHONY: ensure-executable-bin
+ensure-executable-bin:
+	find $(RELEASE_DIR) -type f ! -name "*.sha256" | xargs chmod +x
+
 .PHONY: build-all-platforms
 build-all-platforms: ci-build-linux ci-build-linux-static ci-build-darwin ci-build-darwin-arm64-static ci-build-windows
 
 .PHONY: image-quick
-image-quick:
-	chmod +x $(RELEASE_DIR)/opa_linux_$(GOARCH)*
+image-quick: ensure-executable-bin
 	$(DOCKER) build \
 		-t $(DOCKER_IMAGE):$(VERSION) \
 		--build-arg BASE=gcr.io/distroless/cc \
-		--build-arg BIN=$(RELEASE_DIR)/opa_linux_$(GOARCH) \
+		--build-arg BIN_DIR=$(RELEASE_DIR) \
 		.
 	$(DOCKER) build \
 		-t $(DOCKER_IMAGE):$(VERSION)-debug \
 		--build-arg BASE=gcr.io/distroless/cc:debug \
-		--build-arg BIN=$(RELEASE_DIR)/opa_linux_$(GOARCH) \
+		--build-arg BIN_DIR=$(RELEASE_DIR) \
 		.
 	$(DOCKER) build \
 		-t $(DOCKER_IMAGE):$(VERSION)-rootless \
 		--build-arg USER=1000 \
 		--build-arg BASE=gcr.io/distroless/cc \
-		--build-arg BIN=$(RELEASE_DIR)/opa_linux_$(GOARCH) \
+		--build-arg BIN_DIR=$(RELEASE_DIR) \
 		.
 	$(DOCKER) build \
 		-t $(DOCKER_IMAGE):$(VERSION)-static \
 		--build-arg BASE=gcr.io/distroless/static \
-		--build-arg BIN=$(RELEASE_DIR)/opa_linux_$(GOARCH)_static \
+		--build-arg BIN_DIR=$(RELEASE_DIR) \
+		--build-arg BIN_SUFFIX=_static \
+		.
+
+.PHONY: push-manifest-list-%
+push-manifest-list-%: ensure-executable-bin
+	$(DOCKER) buildx build \
```

Comment:
Before, we built the images and _only pushed them_ in this target. Now, we're building them, aren't we? `docker buildx build` makes me think so. Would it make sense to also change the `image-quick` target to use `docker buildx ...`, and if not, what's the difference? 🤔

Reply:
> Before, we built the images and only pushed them in this target. Now, we're building them, aren't we?

Yes, `buildx build --push` builds and pushes the images. There are now two different make targets for building images:
1. `image-quick` builds the image variants for a single platform, tags them, and loads them to the local docker image store. This target is used by the image smoke tests, which then `docker run` the local tagged images.
2. `push-manifest-list-%` builds and pushes untagged images for multiple platforms, then assembles the image references in a manifest list, tags it, and pushes it. This is used in the release process.

`buildx` does not support loading multi-platform manifests to the docker image store, so we can't use the same build→tag→push workflow that was previously used for single-platform images. It is possible to perform all the same steps `buildx build --push` does in separate commands/targets, but the single command is _significantly_ less complex. I also don't see a reason to break it up.

> Would it make sense to also change the image-quick target to use `docker buildx ...`

We could change `image-quick` to use `buildx build`, but we don't need any Buildx features here. It also requires the extra [`--load` flag](https://github.com/docker/buildx/blob/master/docs/reference/buildx_build.md#-load-the-single-platform-build-result-to-docker-images---load) to get the same result as `docker build`. If you think using `buildx` for everything is more important than the `docker build` simplicity, I can make that change. Either way, the image is built by BuildKit and runnable in the smoke tests.

Reply:
> Either way, the image is built by BuildKit and runnable in the smoke tests.

So if I understand this correctly, it's more a matter of CLI commands used, not about what happens as an effect of those. That's cool, let's keep it. Thanks for taking the time to explain this! 👍

Reply:
That's correct.